### ISSUE TITLE
parser: Evaluate merging TopLevelDecl with Decl to simplify grammar

### ISSUE THREAD

**Issue by mewmew** [mewmew/uc#38]
From [ast.go](https://github.com/mewmew/uc/blob/4bbf81a3bd5931e49a92a104e818306f51545311/ast/ast.go#L36):
```go
// TODO: Evaluate whether TopLevelDecl should be merged with Decl, to simplify
// the structure of the AST; in which case a semantic analysis pass would be
// added to ensure that function declarations are not nested (even if this would
// be an interesting extension to C).
```

**Comment by mewmew** [comment]
After discussions, we've decided to remove the `TopLevelDecl` special case and make function declarations behave like any other declaration. This should simplify the grammar and the AST, while semantic checks should be added to ensure that declarations within functions are not function declarations (unless the GNU extension of nested functions is enabled; see issue #43).

**Comment by mewmew** [comment]
`TopLevelDecl` and `Decl` were merged in commit 3db3068. Closing this issue.
